pkg/service/credential/storage.go

```diff
@@ -80,26 +80,42 @@ type StatusListIndex struct {
 
 func NewCredentialStorage(db storage.ServiceStorage) (*Storage, error) {
 	if db == nil {
-		return nil, errors.New("bolt db reference is nil")
+		return nil, util.LoggingNewError("bolt db reference is nil")
 	}
 
-	randUniqueList := randomUniqueNum(bitStringLength)
-	uniqueNumBytes, err := json.Marshal(randUniqueList)
+	listIndexKeyExists, err := db.Exists(context.Background(), statusListIndexNamespace, currentListIndexKey)
 	if err != nil {
-		return nil, util.LoggingErrorMsg(err, "could not marshal random unique numbers")
+		return nil, util.LoggingErrorMsg(err, "exists for currentListIndexKey")
 	}
 
-	if err := db.Write(context.Background(), statusListIndexNamespace, statusListIndexesKey, uniqueNumBytes); err != nil {
-		return nil, util.LoggingErrorMsg(err, "problem writing status list indexes to db")
+	statusListIndexesKeyExists, err := db.Exists(context.Background(), statusListIndexNamespace, statusListIndexesKey)
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "exists for statusListIndexesKey")
 	}
 
-	statusListIndexBytes, err := json.Marshal(StatusListIndex{Index: 0})
-	if err != nil {
-		return nil, util.LoggingErrorMsg(err, "could not marshal status list index bytes")
+	if listIndexKeyExists != statusListIndexesKeyExists {
+		return nil, util.LoggingNewError("list index and status list index not in the same state")
 	}
 
-	if err := db.Write(context.Background(), statusListIndexNamespace, currentListIndexKey, statusListIndexBytes); err != nil {
-		return nil, util.LoggingErrorMsg(err, "problem writing current list index to db")
+	if !listIndexKeyExists && !statusListIndexesKeyExists {
```

Comment:
nit: consider a utility variable to help make this more readable

like `indexKeysDontExist := !listIndexKeyExists && !statusListIndexesKeyExists`